RecoBTag/DeepFlavour/plugins/DeepFlavourTagInfoProducer.cc

```diff
@@ -0,0 +1,318 @@
+{
+
+  auto output_tag_infos = std::make_unique<DeepFlavourTagInfoCollection>();
+
+  edm::Handle<edm::View<reco::Jet>> jets;
+  iEvent.getByToken(jet_token_, jets);
+
+  edm::Handle<VertexCollection> vtxs;
+  iEvent.getByToken(vtx_token_, vtxs);
+  // reference to primary vertex
+  const auto & pv = vtxs->at(0);
+
+  edm::Handle<SVCollection> svs;
+  iEvent.getByToken(sv_token_, svs);
+
+  edm::Handle<ShallowTagInfoCollection> shallow_tag_infos;
+  iEvent.getByToken(shallow_tag_info_token_, shallow_tag_infos);
+
+  edm::Handle<edm::ValueMap<float>> puppi_value_map;
+  if (use_puppi_value_map_) { 
+    iEvent.getByToken(puppi_value_map_token_, puppi_value_map);
+  }
+
+  edm::Handle<edm::ValueMap<int>> pvasq_value_map;
+  edm::Handle<edm::Association<VertexCollection>> pvas;
+  if (use_pvasq_value_map_) { 
+    iEvent.getByToken(pvasq_value_map_token_, pvasq_value_map);
+    iEvent.getByToken(pvas_token_, pvas);
+  }
+
+  edm::ESHandle<TransientTrackBuilder> track_builder; 
+  iSetup.get<TransientTrackRecord>().get("TransientTrackBuilder", track_builder);
+
+  for (std::size_t jet_n = 0; jet_n <  jets->size(); jet_n++) {
+
+    // create data containing structure
+    deep::DeepFlavourFeatures features;
+
+    // reco jet reference (use as much as possible)
+    const auto & jet = jets->at(jet_n);
+    // dynamical castoting to pointers, null if not possible
+    const auto * pf_jet = dynamic_cast<const reco::PFJet *>(&jet);
+    const auto * pat_jet = dynamic_cast<const pat::Jet *>(&jet);
+    edm::RefToBase<reco::Jet> jet_ref(jets, jet_n);
+    // TagInfoCollection not in an associative container so search for matchs
+    const edm::View<reco::ShallowTagInfo> & taginfos = *shallow_tag_infos;
+    edm::Ptr<reco::ShallowTagInfo> match;
+    // Try first by 'same index'
+    if ((jet_n < taginfos.size()) && (taginfos[jet_n].jet() == jet_ref)) {
+        match = taginfos.ptrAt(jet_n);
+    } else {
+      // otherwise fail back to a simple search
+      for (auto itTI = taginfos.begin(), edTI = taginfos.end(); itTI != edTI; ++itTI) {
+        if (itTI->jet() == jet_ref) { match = taginfos.ptrAt( itTI - taginfos.begin() ); break; }
+      }
+    }
+    reco::ShallowTagInfo tag_info;
+    if (match.isNonnull()) {
+      tag_info = *match; 
+    } // will be default values otherwise
+
+    // fill basic jet features
+    deep::jet_features_converter(jet, features.jet_features);
+
+    // fill number of pv
+    features.npv = vtxs->size();
+
+    // fill features from ShallowTagInfo
+    const auto & tag_info_vars = tag_info.taggingVariables();
+    btag_features_converter(tag_info_vars, features.tag_info_features);
+
+    // copy which will be sorted
+    auto svs_sorted = *svs;     
+    // sort by dxy
+    std::sort(svs_sorted.begin(), svs_sorted.end(),
+              [&pv](const auto & sva, const auto &svb)
+              { return deep::sv_vertex_comparator(sva, svb, pv); });
```

Comment:
the sort apparently does not depend on the jet. Please sort outside of the jet loop.